src/components/student/AssignmentsStudentView.js

```diff
@@ -8,13 +9,92 @@ import React, {useState} from 'react';
 // display a table with columns  Course Id, Assignment Title, Assignment DueDate, Score
 
 const AssignmentsStudentView = (props) => {
-    
-     
-    return(
-        <> 
-            <h3>Not implemented</h3>   
-        </>
+
+    const studentId = 3; // Placeholder until login is implemented
+    const headers = ['Course', 'Title', 'Due Date', 'Score'];
+
+    // States
+    const [assignments, setAssignments] = useState([]);
+    const [showHeaders, setShowHeaders] = useState(false);
+    const [message, setMessage] = useState('');
+    const [search, setSearch] = useState({ studentId: studentId, year: '', semester: '' });
+
+    // Function to fetch assignments data from the server
+    const fetchAssignments = async () => {
+        try {
+            const response = await fetch(`${SERVER_URL}/assignments?studentId=${search.studentId}&year=${search.year}&semester=${search.semester}`);
+            if (response.ok) {
+                const data = await response.json();
+                setAssignments(data);
```

Comment:
Might want to setAssignments to empty if the query doesn't return any assignments.